mantidproject/mantid
# Match Osiris script with the indirect diffraction UI

**Description of work.**
This PR matches the Osiris script code with the Indirect Diffraction UI code for consistency, Previously, the script and the UI produce different results. Now, using the two methods the user will get the same results. This was done by making the following changes to the scripts:
1- adding ApplyDiffCal call before converting units.
2- swapping the order of the ConvertUnits and DiffractionFocussing algorithms applications.
3- adding ReplaceSpecialValues after Divide when correcting the sample run using a vanadium run.
4- fix a rebinning problem when merging runs across different dspacing ranges.
5- rebin and average spectra before applying the focusing.

Fix 4 was also applied to the Indirect Diffraction UI. Logic to remove any input workspaces that had the same x range as another workspace from the same dRange was also removed from the Indirect Diffraction UI

**To test:**
1- Open the Indirect Diffraction UI
2- Add a sample run, calibration file, and a vanadium run
3- Run the UI
4- Repeat 1-2 steps using the Osiris script
5- Plot the data from the script and the UI and they should be exactly similar

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
